src/bos/operators/utils/clients/bos/base.py

```diff
@@ -121,3 +122,76 @@ def delete_items(self, **kwargs):
             return json.loads(response.text)
         else:
             return None
+
+
+class BaseBosTenantAwareEndpoint(BaseBosEndpoint):
+    """
+    This base class provides generic access to the BOS API for tenant aware endpoints.
+    The individual endpoint needs to be overridden for a specific endpoint.
+    """
+
+    @log_call_errors
+    def get_item(self, item_id, tenant):
+        """Get information for a single BOS item"""
+        url = self.base_url + '/' + item_id
+        session = requests_retry_session()
+        response = session.get(url, headers=get_new_tenant_header(tenant))
+        response.raise_for_status()
+        item = json.loads(response.text)
+        return item
+
+    @log_call_errors
+    def get_items(self, **kwargs):
+        """Get information for all BOS items"""
+        session = requests_retry_session()
```

Comment:
It's silly to re-instantiate this with each method call. We should move it to __init__.